src/cobo/handshake.c

```diff
@@ -327,16 +344,65 @@ static int handshake_main(int sockfd, handshake_protocol_t *hdata, uint64_t sess
    /**
     * Recieve a packet_buffer on the network
     **/
+   result = recv_packet(sockfd, &recvd_long_packet_buffer, &recvd_long_packet_buffer_size);
+   if (result < 0) {
+      debug_printf("Problem receiving packet\n");
+      return_result = result;
+      socket_error = 1;
+      goto done;
+   }
+
+   /**
+    * Decrypt their packet with their random number
+   **/
+   debug_printf("Decrypting their random number packet\n");
+   result = decrypt_packet(hdata, &recvd_packet, sizeof(handshake_packet_t), recvd_long_packet_buffer, recvd_long_packet_buffer_size);
+   if (result < 0) {
+      debug_printf("Error decrypting and checking received packet\n");
+      return_result = result;
+      goto done;
+   }
+
```

Comment:
See comment below about splitting up compare_packet(). The compare_handshake_packet() would go here.